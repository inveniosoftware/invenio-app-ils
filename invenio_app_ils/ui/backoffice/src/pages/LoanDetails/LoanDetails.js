import React, { Component } from 'react';
import { Loader, Message } from 'semantic-ui-react';
import { LoanTitle } from './components/LoanTitle/LoanTitle';
import { LoanMetadata } from './components/LoanMetadata/LoanMetadata';
import { LoanActions } from './components/LoanActions/LoanActions';

export default class LoanDetails extends Component {
  constructor(props) {
    super(props);
    this.fetchLoanDetails = this.props.fetchLoanDetails;
    this.postLoanAction = this.props.postLoanAction;
  }

  componentDidMount() {
    this.unlisten = this.props.history.listen((location, action) => {
      if (location.state && location.state.recid) {
        this.fetchLoanDetails(location.state.loanId);
      }
    });
    this.fetchLoanDetails(this.props.match.params.loanId);
  }

  componentWillUnmount() {
    this.unlisten();
  }

  render() {
    let {
      data,
      error,
      fetchLoading,
      actionLoading,
      loanActionError,
    } = this.props;
    if (fetchLoading) return <Loader active inline="centered" />;
    if (loanActionError) {
      return (
        <Message
          icon="exclamation"
          header="Oups, something went horribly wrong!"
          content={error.message}
        />
      );
    }
    console.log(data);
    return (
      <section>
        <LoanTitle loanId={data.metadata.loan_pid} />
        <LoanMetadata data={data} />
        <LoanActions
          data={data.metadata}
          actions={data.availableActions}
          actionLoading={actionLoading}
          onAction={this.postLoanAction}
        />
      </section>
    );
  }
}
