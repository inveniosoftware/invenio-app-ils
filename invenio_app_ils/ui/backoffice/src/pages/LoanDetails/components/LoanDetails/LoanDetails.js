import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Container } from 'semantic-ui-react';
import { compose } from 'redux';
import { withError, withLoader } from 'common/hoc';
import { LoanMetadata } from '../LoanMetadata/LoanMetadata';
import { LoanActions } from '../LoanActions/LoanActions';

class LoanDetails extends Component {
  render() {
    let { data, onAction } = this.props;
    return (
      <Container>
        <LoanMetadata data={data} />
        <LoanActions data={data} onAction={onAction} />
      </Container>
    );
  }
}

LoanDetails.propTypes = {
  data: PropTypes.object.isRequired,
};

export default compose(
  withLoader,
  withError
)(LoanDetails);
