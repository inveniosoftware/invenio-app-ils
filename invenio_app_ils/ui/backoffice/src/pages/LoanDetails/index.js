import { compose } from 'redux';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { fetchLoanDetails, postLoanAction } from './state/actions';
import LoanDetailsContainerComponent from './LoanDetailsContainer';

const mapStateToProps = state => ({
  data: state.loanDetails.data,
  error: state.loanDetails.error,
  isLoading: state.loanDetails.isLoading,
  actionLoading: state.loanDetails.actionLoading,
  loanActionError: state.loanDetails.loanActionError,
});

const mapDispatchToProps = dispatch => ({
  fetchLoanDetails: loanId => dispatch(fetchLoanDetails(loanId)),
  postLoanAction: (url, data) => dispatch(postLoanAction(url, data)),
});

export const LoanDetailsContainer = compose(
  withRouter,
  connect(
    mapStateToProps,
    mapDispatchToProps
  )
)(LoanDetailsContainerComponent);
