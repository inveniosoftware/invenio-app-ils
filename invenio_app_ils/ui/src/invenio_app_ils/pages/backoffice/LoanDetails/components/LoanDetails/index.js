import { compose } from 'redux';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import LoanDetailsComponent from './LoanDetails';

const mapStateToProps = state => ({
  isLoading: state.loanDetails.isLoading,
  data: state.loanDetails.data,
  error: state.loanDetails.error,
  hasError: state.loanDetails.hasError,
});

export const LoanDetails = compose(
  withRouter,
  connect(mapStateToProps)
)(LoanDetailsComponent);
