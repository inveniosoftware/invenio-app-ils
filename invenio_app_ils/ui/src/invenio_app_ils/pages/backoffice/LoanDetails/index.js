import { compose } from 'redux';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { fetchLoanDetails } from './state/actions';
import LoanDetailsContainerComponent from './LoanDetailsContainer';

const mapDispatchToProps = dispatch => ({
  fetchLoanDetails: loanPid => dispatch(fetchLoanDetails(loanPid)),
});

export const LoanDetailsContainer = compose(
  withRouter,
  connect(
    null,
    mapDispatchToProps
  )
)(LoanDetailsContainerComponent);
