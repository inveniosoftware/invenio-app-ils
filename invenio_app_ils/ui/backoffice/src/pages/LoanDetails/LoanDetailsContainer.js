import React, { Component } from 'react';
import LoanDetails from './components/LoanDetails/LoanDetails';

export default class LoanDetailsContainer extends Component {
  constructor(props) {
    super(props);
    this.fetchLoanDetails = this.props.fetchLoanDetails;
    this.postLoanAction = this.props.postLoanAction;
  }

  componentDidMount() {
    this.unlisten = this.props.history.listen((location, action) => {
      if (location.state && location.state.loanid) {
        this.fetchLoanDetails(location.state.loanId);
      }
    });
    this.fetchLoanDetails(this.props.match.params.loanId);
  }

  componentWillUnmount() {
    this.unlisten();
  }

  render() {
    return <LoanDetails {...this.props} />;
  }
}
