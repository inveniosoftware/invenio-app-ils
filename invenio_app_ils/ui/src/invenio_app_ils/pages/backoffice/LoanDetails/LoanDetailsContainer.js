import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Container } from 'semantic-ui-react';
import { LoanDetails } from './components';

export default class LoanDetailsContainer extends Component {
  constructor(props) {
    super(props);
    this.fetchLoanDetails = this.props.fetchLoanDetails;
  }

  componentDidMount() {
    this.unlisten = this.props.history.listen(location => {
      if (location.state && location.state.loanPid) {
        this.fetchLoanDetails(location.state.loanPid);
      }
    });
    this.fetchLoanDetails(this.props.match.params.loanPid);
  }

  componentWillUnmount() {
    this.unlisten();
  }

  render() {
    return (
      <Container>
        <LoanDetails />
      </Container>
    );
  }
}

LoanDetailsContainer.propTypes = {
  fetchLoanDetails: PropTypes.func.isRequired,
};
