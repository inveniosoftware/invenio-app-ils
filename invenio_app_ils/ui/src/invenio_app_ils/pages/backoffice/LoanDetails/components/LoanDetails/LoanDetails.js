import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Container } from 'semantic-ui-react';
import { Loader, Error } from '../../../../../common/components';
import { LoanMetadata } from '../';
import { LoanActions } from '../';

export default class LoanDetails extends Component {
  render() {
    const {
      isLoading,
      isActionLoading,
      data,
      hasError,
      actionHasError,
    } = this.props;
    const errorData = hasError || actionHasError ? data : null;
    const anyIsLoading = isLoading || isActionLoading;
    return (
      <Loader isLoading={anyIsLoading}>
        <Error error={errorData}>
          <Container>
            <LoanMetadata />
            <LoanActions />
          </Container>
        </Error>
      </Loader>
    );
  }
}

LoanDetails.propTypes = {
  isLoading: PropTypes.bool.isRequired,
  isActionLoading: PropTypes.bool.isRequired,
  data: PropTypes.object,
  hasError: PropTypes.bool.isRequired,
  actionHasError: PropTypes.bool.isRequired,
};
