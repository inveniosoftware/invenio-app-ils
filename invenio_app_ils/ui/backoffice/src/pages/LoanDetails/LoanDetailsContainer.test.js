import React from 'react';
import { mount } from 'enzyme';
import { Message } from 'semantic-ui-react';
import LoanDetailsContainer from './LoanDetailsContainer';
import LoanDetails from './components/LoanDetails/LoanDetails';

const defaultProps = {
  data: {
    loan_pid: 42,
    metadata: {
      title: 'title',
    },
    availableActions: {
      action: 'action',
    },
  },
  isLoading: true,
};

describe('LoanDetailsContainer', () => {
  let component;

  beforeAll(() => {
    LoanDetailsContainer.prototype.componentDidMount = () => {};
    LoanDetailsContainer.prototype.componentWillUnmount = () => {};
  });

  afterEach(() => {
    component.unmount();
  });

  it('should render correctly props', () => {
    component = mount(<LoanDetailsContainer {...defaultProps} />);
    expect(component).toMatchSnapshot();
  });

  it('should render an error when fetch fails', () => {
    let props = {
      ...defaultProps,
      isLoading: false,
      error: {
        message: 'error',
      },
    };
    component = mount(<LoanDetailsContainer {...props} />);
    expect(component).toMatchSnapshot();
    const message = component.find(Message);
    expect(message.exists()).toEqual(true);
  });

  it('should render loan details when fetch succeeds', () => {
    let props = {
      ...defaultProps,
      isLoading: false,
    };
    component = mount(<LoanDetailsContainer {...props} />);
    expect(component).toMatchSnapshot();
    const detailsComponent = component.find(LoanDetails);
    expect(detailsComponent.exists()).toEqual(true);
    const message = component.find(Message);
    expect(message.exists()).toEqual(false);
  });
});
