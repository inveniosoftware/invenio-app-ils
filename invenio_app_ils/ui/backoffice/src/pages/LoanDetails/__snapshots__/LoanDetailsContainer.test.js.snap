// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoanDetailsContainer should render an error when fetch fails 1`] = `
<LoanDetailsContainer
  data={
    Object {
      "availableActions": Object {
        "action": "action",
      },
      "loan_pid": 42,
      "metadata": Object {
        "title": "title",
      },
    }
  }
  error={
    Object {
      "message": "error",
    }
  }
  isLoading={false}
>
  <WithLoader
    data={
      Object {
        "availableActions": Object {
          "action": "action",
        },
        "loan_pid": 42,
        "metadata": Object {
          "title": "title",
        },
      }
    }
    error={
      Object {
        "message": "error",
      }
    }
    isLoading={false}
  >
    <WithError
      data={
        Object {
          "availableActions": Object {
            "action": "action",
          },
          "loan_pid": 42,
          "metadata": Object {
            "title": "title",
          },
        }
      }
      error={
        Object {
          "message": "error",
        }
      }
    >
      <ApiError
        error={
          Object {
            "message": "error",
          }
        }
      >
        <Message
          content="error"
          header="Oups, failed to fetch data from API!"
          icon="exclamation"
        >
          <div
            className="ui icon message"
          >
            <Icon
              as="i"
              name="exclamation"
            >
              <i
                aria-hidden="true"
                className="exclamation icon"
              />
            </Icon>
            <MessageContent>
              <div
                className="content"
              >
                <MessageHeader
                  content="Oups, failed to fetch data from API!"
                >
                  <div
                    className="header"
                  >
                    Oups, failed to fetch data from API!
                  </div>
                </MessageHeader>
                <p>
                  error
                </p>
              </div>
            </MessageContent>
          </div>
        </Message>
      </ApiError>
    </WithError>
  </WithLoader>
</LoanDetailsContainer>
`;

exports[`LoanDetailsContainer should render correctly props 1`] = `
<LoanDetailsContainer
  data={
    Object {
      "availableActions": Object {
        "action": "action",
      },
      "loan_pid": 42,
      "metadata": Object {
        "title": "title",
      },
    }
  }
  isLoading={true}
>
  <WithLoader
    data={
      Object {
        "availableActions": Object {
          "action": "action",
        },
        "loan_pid": 42,
        "metadata": Object {
          "title": "title",
        },
      }
    }
    isLoading={true}
  >
    <ApiLoader>
      <Loader
        active={true}
        inline="centered"
        size="huge"
      >
        <div
          className="ui huge active centered inline loader"
        />
      </Loader>
    </ApiLoader>
  </WithLoader>
</LoanDetailsContainer>
`;

exports[`LoanDetailsContainer should render loan details when fetch succeeds 1`] = `
<LoanDetailsContainer
  data={
    Object {
      "availableActions": Object {
        "action": "action",
      },
      "loan_pid": 42,
      "metadata": Object {
        "title": "title",
      },
    }
  }
  isLoading={false}
>
  <WithLoader
    data={
      Object {
        "availableActions": Object {
          "action": "action",
        },
        "loan_pid": 42,
        "metadata": Object {
          "title": "title",
        },
      }
    }
    isLoading={false}
  >
    <WithError
      data={
        Object {
          "availableActions": Object {
            "action": "action",
          },
          "loan_pid": 42,
          "metadata": Object {
            "title": "title",
          },
        }
      }
    >
      <LoanDetails
        data={
          Object {
            "availableActions": Object {
              "action": "action",
            },
            "loan_pid": 42,
            "metadata": Object {
              "title": "title",
            },
          }
        }
      >
        <Container>
          <div
            className="ui container"
          >
            <LoanMetadata
              data={
                Object {
                  "availableActions": Object {
                    "action": "action",
                  },
                  "loan_pid": 42,
                  "metadata": Object {
                    "title": "title",
                  },
                }
              }
            >
              <Segment>
                <div
                  className="ui segment"
                >
                  <Grid>
                    <div
                      className="ui grid"
                    >
                      <GridColumn>
                        <div
                          className="column"
                        >
                          <h1>
                            Loan - 
                          </h1>
                          <Form
                            as="form"
                          >
                            <form
                              className="ui form"
                              onSubmit={[Function]}
                            >
                              <FormField
                                key="title"
                                name="loan-field"
                              >
                                <div
                                  className="field"
                                  name="loan-field"
                                >
                                  <label
                                    className="field-name"
                                  >
                                    title
                                  </label>
                                  <p>
                                    title
                                  </p>
                                </div>
                              </FormField>
                            </form>
                          </Form>
                        </div>
                      </GridColumn>
                    </div>
                  </Grid>
                </div>
              </Segment>
            </LoanMetadata>
            <LoanActions
              data={
                Object {
                  "availableActions": Object {
                    "action": "action",
                  },
                  "loan_pid": 42,
                  "metadata": Object {
                    "title": "title",
                  },
                }
              }
            >
              <List
                horizontal={true}
              >
                <div
                  className="ui horizontal list"
                  role="list"
                >
                  <ListItem
                    key="action"
                  >
                    <div
                      className="item"
                      onClick={[Function]}
                      role="listitem"
                    >
                      <Button
                        as="button"
                        onClick={[Function]}
                        primary={true}
                        role="button"
                      >
                        <button
                          className="ui primary button"
                          onClick={[Function]}
                          role="button"
                        >
                          action
                        </button>
                      </Button>
                    </div>
                  </ListItem>
                </div>
              </List>
            </LoanActions>
          </div>
        </Container>
      </LoanDetails>
    </WithError>
  </WithLoader>
</LoanDetailsContainer>
`;
